workflow/artifacts/s3/s3.go

```diff
@@ -37,7 +39,12 @@ type ArtifactDriver struct {
 
 var (
 	_            artifactscommon.ArtifactDriver = &ArtifactDriver{}
-	defaultRetry                                = wait.Backoff{Duration: time.Second * 2, Factor: 2.0, Steps: 5, Jitter: 0.1}
+	defaultRetry                                = wait.Backoff{
```

Comment:
I don't think this will work, because these enviers are never set.

Reply:
Could you propose new hard-coded values? Presumably more retries?

Reply:
This way we are able to set the retry in a way that matches our situation.

Reply:
I recall that we had some logic on this on an `init()` somewhere. Perhaps see if you can reuse those instead.

Reply:
@terrytangyuan and @alexec I propose to use `DefaultRetry` from `util/retry/retry.go:18` 
except that `DefaultRetry` doesn't have jitter. Is it ok to add jitter to `DefaultRetry`? Or rather not?

Reply:
Or does `DefaultRetry` fit the bill beter / or perhaps even its own set of env parameters since you might see this as a special case (like `DefaultRetry` and `ExecutorRetry`)?

Reply:
whatever we do, please log the settings

Reply:
Definitely not S3 specific retries. ExecutorRetry sounds good. Perhaps this PR can be generalized to other artifact drivers? WDYT @alexec

Reply:
I think we should be able to set-up different one for different cases. But, I wonder if we want to do more retries for artifacts by default? E.g. spend up-to 1m trying to get it?

Reply:
The current retries have been in the code for a while. This was the last time they were increased: https://github.com/argoproj/argo-workflows/pull/1138 which fixes this issue: https://github.com/argoproj/argo-workflows/issues/1081, there is a difference between the two: 10 milliseconds (or less than a second) and the current 2 seconds (a bit over a minute). For our case (large workflows with a lot of steps that take a long time to process) waiting a bit for a transient errors is ok and I even tweaked them higher so both backoffs are even a bit too small for our case.

Reply:
@alexec and @terrytangyuan what do you think about using `ExecutorRetry` as default? I somehow don't feel I should make this decision. I'm quite agnostic to what default is used, apart from my thoughts above ☝️. 
If you decide I'll make a commit that contains that and:

> whatever we do, please log the settings

Reply:
The idea of using the ExecutorRetry creates a cyclical import.

Reply:
I propose this solution: https://github.com/PDOK/argo-workflows/pull/3